### PR TITLE
DCC Issue 706 - REST API V1 /api/v1/plans is returning publicly visible

### PR DESCRIPTION
**Pull request by johnpinto1**
plans from other Orgs.

Fix for DCC Issue https://github.com/DigitalCurationCentre/DMPonline-Service/issues/706

Changes to app/policies/api/v1/plans_policy.rb:
- In method resolve() removed
   ids = Plan.publicly_visible.pluck(:id)
- In method initialize(client, plan) added missing variable
   @user = client

Note: the call  /api/v1/plans  no longer returns public plans from other orgs:

curl -k -v -H "Accept: application/json" -H "Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJjbGllbnRfaWQiOiJqYUBrdGguc2UiLCJleHAiOjE2NDY5OTIxODB9.pQeMb3uIfLiI-Ge2hPUcM04xOcrfo2Ld4apELqYqMNE"  http://localhost:3000/api/v1/plans | grep -A3 affiliation | grep name

         "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",
            "name": "KTH Royal Institute of Technology",

### CODE REVIEW

File: app/policies/api/v1/plans_policy.rb [hunk 1]
```diff
@@ -16,9 +16,8 @@ class Scope
         # User (non-admin) can view: any personal or organisationally_visible
         # User (admin) can view: all from users of their organisation
         def resolve
-          ids = Plan.publicly_visible.pluck(:id)
-          ids += plans_for_client if @user.is_a?(ApiClient)
-          ids += plans_for_user if @user.is_a?(User)
+          ids = plans_for_client if @user.is_a?(ApiClient)
```
Comment (briri) [inline app/policies/api/v1/plans_policy.rb:19]: lets consolidate these to: `ids = @user.is_a?(ApiClient) ? plans_for_client : plans_for_user`
and then a simple `return [] unless @user.present?` in those two 'plans_for_x' methods to be safe

File: app/policies/api/v1/plans_policy.rb [hunk 2]
```diff
@@ -40,6 +39,7 @@ def plans_for_user
         def initialize(client, plan)
           super()
           @client = client
```
[the remark this answers is not in the document]
Comment (johnpinto1) [reply app/policies/api/v1/plans_policy.rb]: @briri  Just removed @client = client as requested.

File: app/policies/api/v1/plans_policy.rb (hunk 1 above)
Comment (johnpinto1) [reply app/policies/api/v1/plans_policy.rb:19]: Will make change soon.

Comment (johnpinto1) [reply app/policies/api/v1/plans_policy.rb]: @briri updated with change requested.

**Pull request merged**